patch/save_patches_old.txt

```diff
@@ -0,0 +1,578 @@
+	call GetGameDataPointer
+	seek #result
+	pop #result
+	seekfwd 4
+
+	;we first need to check the first four key item bytes that were overwritten
+	pop #temp
+.key_items_loop_1
+	and #1, #temp, 0xff
+	call .loop_common_get
+	jumpz #1, .key_items_loop_1 ; quietly ignore invalid/obsolete indexes
+	ifeq #1, 0xff, .done_loop
+	decrement #1 ; loop_common_get returns a valid index from 1-1c, we want 0-1b now
+	push #temp
+	call .loop_common_set
+	pop #temp
+	shiftright #temp, 8
+	jumpz #temp, .key_items_loop_2
+	jump .key_items_loop_1
+
+	;now we can check the remaining key items
+.key_items_loop_2
+	readbyte #1
+	call .loop_common_get
+	jumpz #1, .key_items_loop_2
+	ifeq #1, 0xff, .done_loop
+	decrement #1
+	call .loop_common_set
+	jump .key_items_loop_2
+
+.done_loop
+	; shift data from NumPCItems to before RegisteredItems
+	set #1, NUM_PC_ITEMS_OFFSET_v1
+	call GetGameDataPointer
+	seek #result
+	set #2, 26
+	set #1, 1682
+	call ShiftSaveDataBack 
+
+	; update registered items, quietly clear items that are out of range (i.e. Repel)
+	set #1, REG_KEY_ITEM_FLAGS_v1
+	call GetGameDataPointer
+	seek #result
+	writebyte #zero ; clear wRegisteredItemFlags
+	pushpos
+
+	set #1, REG_KEY_ITEMS_v0
+	call GetGameDataPointer
+	seek #result
+
+	set #temp, 4
+.registered_loop
+	getfilebyte #1
+	call .loop_common_get
+	jumpz #1, .clear
+	increment #1
+	jumpz #1, .clear ; also clears 0xff
+	decrement #1
+.clear
+	writebyte #1
+	decrement #temp
+	jumpnz #temp, .registered_loop
+
+	; shift data from RegisteredItems to end of save bank
+	; (obviously needs to check which bank is being affected)
+	poppos
+	set #2, 25
+	set #1, 5275
+	ifeq #savefile, 1, .shift
+	set #1, 667
+	call ShiftSaveDataBack
+
+	; we don't care about overwriting game data with box data
+	; if we're loading from a backup anyway
+	seek sBox_v1
+	set #1, 4899
+.shift
+	jump ShiftSaveDataBack
+
+; ==== common key item loop functions ====
+.loop_common_get
+; gets the appropriate adjusted flag value for key item loops
+; returns #1 with the result (0xff if terminator is reached)
+	ifeq #1, 0xff, .continue
+	ifeq #1, 0x5b, .zero ; ignore EXP_SHARE key item
+	ifle #1, 0x54, .zero
+	ifge #1, 0x72, .zero
+	ifle #1, 0x5a, .sub0x54
+	ifge #1, 0x70, .sub0x54
+	decrement #1 ; subtract #temp, 0x55
+.sub0x54
+	subtract #1, 0x54
+	return
+.zero
+	set #1, #zero
+.continue
+	return
+
+.loop_common_set
+; sets the appropriate key item flag
+	pushpos
+	seek #result
+	push #result
+	set #2, 1
+	call FlagAction
+	pop #result
+	poppos
+	return
+
+; ==== resize pockets ====
+; positive differences shift forward, negative differences shift backwards
+; #r0: OLD build number
+; #1:  item pocket size difference
+; #2:  medicine pocket size difference
+; #3:  balls pocket size difference
+; #4:  berries pocket size difference
+; exits with error if build number is out of expected range
+ResizePocketsToBuild4:
+	set #address, .pocket_patch_table
+.patch_loop
+	gethalfwordinc #r0, #address
+	ifeq #r0, 0xff, .error
```

Comment:
Are you sure this is correct, given that the build number you just read is a halfword?